src/atlannot/_atlalign.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
```

Comment:
- Do you mind adding the header for the LICENSE ?
- Also, do you think it is better to keep this under `_atlalign` module ? Is there a specific reason to put `_` at the start of the name of the module ? I understand that it is because the code is very similar to `atlalign` but is it not confusing from someone outside of our team ?

Reply:
Regarding the second point, I was not really sure what the best setup could be. IMO what is really confusing is that currently that are a lot of functions that have the `iou` in their name.

See below a list of them
* `iou_score_single` - https://github.com/BlueBrain/atlas-annotation/blob/4becc1389b3126527b2081b929caf3f10e66287c/src/atlannot/_atlalign.py#L6
* `iou_score` - https://github.com/BlueBrain/atlas-annotation/blob/4becc1389b3126527b2081b929caf3f10e66287c/src/atlannot/_atlalign.py#L71
* `iou` - https://github.com/BlueBrain/atlas-annotation/blob/4becc1389b3126527b2081b929caf3f10e66287c/src/atlannot/evaluation.py#L105



I am open for suggestoins!

Reply:
I totally agree. Maybe we should rename the last one (because it is taking the hierarchy of region labels into account). However, we can let it the way it is for now.

Reply:
Licences done in 9ad3ee9

IMO we can let it the way it is right now:)